pkg/server/resource_lister.go

```diff
@@ -37,11 +41,15 @@ func (l *ResourceLister) handleResources(w http.ResponseWriter, r *http.Request)
 
 	if resp.StatusCode != http.StatusOK {
 		err := fmt.Errorf("console service account cannot list resource: %s", resp.Status)
-		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: err.Error()})
+		serverutils.SendResponse(w, resp.StatusCode, serverutils.ApiError{Err: err.Error()})
 		return
 	}
 
 	w.WriteHeader(resp.StatusCode)
-	io.Copy(w, resp.Body)
+	if l.ResponseFilter != nil {
+		l.ResponseFilter(w, resp.Body)
+	} else {
+		io.Copy(w, resp.Body)
+	}
```

Comment:
Why don't you do the following instead:
```
type ResourceLister interface {
    HandleResources(http.ResponseWriter, io.Reader)
}

type resourceLister struct {
	bearerToken    string
	requestURL  *url.URL
	client      *http.Client
	responseFilter FilterFunction
}

func NewResourceLister(bearerToken string, requestURL *url.URL, client *http.Client, respFilter FilterFunction) *ResourceLister {
    r := &resourceLister{
        bearerToken: bearerToken,
        requestURL: requestURL,
        client: client,
        responseFilter: respFilter,
    }
    if r.responseFilter == nil {
        r.responseFilter = io.Copy
    }
    return r
}
```
to prevent this runtime check?

Reply:
What about this one ?

Reply:
Tried adding above factory pattern but `r.responseFilter = io.Copy` assignment enforces FilterFunction to have same declaration as io.Copy() & we end up with more type casting & additional code. Is it worth doing this considering the fact that this endpoint is a workaround for now & will be removed when we have [knative/eventing#2879](https://github.com/knative/eventing/issues/2879).

Reply:
How much more, and do you have a plan for how long this workaround will be in place?
Would be good to know how temporary this will be.

Reply:
Is it more than just wrapping io.Copy() with error handler writing error in the response?
Note that currently you’re discarding the error of io.Copy(), which is wrong

Reply:
@stlaz handled error of io.Copy() as well as added factory pattern. PTAL
@benjaminapetersen There is an open request to provide better support here: knative/eventing#2879. Once we have an alternative solution, we will remove this service.